scaladoc/src/dotty/tools/scaladoc/site/StaticSiteLoader.scala

```diff
@@ -113,11 +113,37 @@ class StaticSiteLoader(val root: File, val args: Scaladoc.Args)(using StaticSite
     StaticSiteRoot(withBlog, mappings)
   }
 
+  var hiddenBlog = false
+
+  def readYml: (Option[Boolean], Option[Boolean], Option[Boolean], String) =
+    val ymlPath = root.toPath.resolve("blog.yml")
+    if (Files.exists(ymlPath)) then
+      val yamlContent = Source.fromFile(ymlPath.toString).getLines().mkString("\n")
+      val hidden = if (yamlContent.contains("hidden: true")) Some(true) else None
+      val input = if (yamlContent.contains("input:")) Some(true) else None
+      val output = if (yamlContent.contains("output:")) Some(true) else None
+      (hidden, input, output, yamlContent)
+    else
+      (None, None, None, "")
+
   def loadBlog(): Option[LoadedTemplate] = {
+    val (hidden, input, output, yamlContent) = readYml
+    val lines = yamlContent.split("\n")
+    val rootPath = input.collect {
+      case true =>
+        lines.collectFirst { case line if line.contains("input:") => line.replaceFirst("input:", "").trim }
+          .map(ctx.resolveNewBlogPath)
+          .getOrElse(ctx.blogPath)
+    }.getOrElse(ctx.blogPath)
+    val defaultDirectory = output.collect {
+    case true =>
+      lines
+        .collectFirst { case line if line.contains("output:") => line.replaceFirst("output:", "").trim }
+        .getOrElse("blog")
+    }.getOrElse("blog")
```

Comment:
I suggest using actual Yaml parser for parsing yaml. We already have `jackson` library in scaladoc, some usage examples can be seen in `SidebarParser.scala`.